src/layout/Header/Header.tsx

```diff
@@ -1,34 +1,114 @@
-import React from "react";
-import { Link } from "react-router-dom";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Link, useHistory } from "react-router-dom";
 import "./style.css";
 import { useAppSelector } from "../../hooks/useAppSelector";
 import { Routes } from "../../constants/Routes";
+import { useAppDispatch } from "../../hooks/useAppDispatch";
+import { logout } from "../../store/reducers/actionCreator";
+import Loader from "../../components/Loader/Loader";
+import { useOutsideClick } from "../../hooks/useOutsideClick";
 
 const Header = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const history = useHistory();
+
+  const dispatch = useAppDispatch();
+  const { email } = useAppSelector((state) => state.authReducer);
+  const { isActivated, isAuth } = useAppSelector((state) => state.authReducer);
+  const Logout = () => {
+    dispatch(logout());
+  };
+
+  function getFirstLetter(email: string) {
+    if (!email) return email;
+
+    return email[0].toUpperCase();
+  }
+  const { loading } = useAppSelector((state) => state.authReducer);
+
   const { favouriteMovies } = useAppSelector(
     (state) => state.movieFavouriteReducer
   );
+
+  const popUpRef = useRef();
+
+  useOutsideClick(popUpRef, () => {
+    setModalOpen(false);
+  });
+
   return (
-    <div className="header">
-      <div className="header__container">
-        <h2 className="header__title">
-          MOVIES <span>SEARCH</span>
-        </h2>
-        <nav className="header__nav">
-          <ul className="header__nav-links">
-            <li>
-              <Link to={Routes.HOME}>Home</Link>
-            </li>
-            <li>
-              <Link to={Routes.FAVOURITES}>Favourites</Link>
-              <span className="header__link-county">
-                {favouriteMovies.length}
-              </span>
-            </li>
-          </ul>
-        </nav>
+    <>
+      {loading && <Loader></Loader>}
+      <div className="header">
```

Comment:
Try to code split this component into small component. For example: NavMenu, UserPersona